cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -371,13 +373,23 @@ public void error(Throwable cause) {
           Closeables.closeQuietly((Closeable) programRunner);
         }
       }
+
```

Comment:
This method becomes too long. Please refactor into some smaller submethods

Reply:
Did some refactoring to split it.